### ISSUE TITLE
idf.py writing to multiple serial ports in parallel (IDFGH-12884)

### ISSUE THREAD

**Issue by Teesmo** [espressif/esp-idf#13846]
Is parallelization still the only/best way to achieve this? Or is there an idf.py implementation that can now accomplish this?

  >             Hi @josep112 ,
> 
> We don't have this feature at the moment, however we're considering how to expose this kind of functionality.
> 
> Because esptool.py is a command line tool, it's easy to combine it with other tools in order to get this kind of parallel execution, though.
> 
> For example, as you seem to be on Linux, you can use [GNU Parallel](https://www.gnu.org/software/parallel/):
> 
> ```
> parallel --group esptool.py --port '{}' write_flash 0x0 blink.bin ::: /dev/ttyUSB0 /dev/ttyUSB1 /dev/ttyUSB2
> ```
> 
> This will run esptool.py in parallel on the three serial ports named after the `:::`. The output is grouped per-command, and no output is printed until each esptool.py finishes (otherwise output is interleaved and can be very messy and hard to understand.)
> 
> Add the `--dry-run` argument to parallel for easier debugging. Consult the Parallel docs for more details.
> 
> You can also store the list of serial ports in a file, or pipe them in.
> 
> _Originally posted by @projectgus in https://github.com/espressif/esptool/issues/227#issuecomment-325242160_
>

**Comment by dobairoland** [comment]
The situation is the same as it was back then. And I don't think we will address this given that command line tools can be interconnected, and is best to have the functionality separated, each tool focus on its area-of-scope, instead of making huge tools with out-of-scope functionality (reading e-mails, publishing to web, and similar).
